p2p/net/swarm/swarm_listen.go

```diff
@@ -13,58 +14,56 @@ import (
 	manet "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net"
 	ps "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-peerstream"
 	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
-	multierr "github.com/ipfs/go-ipfs/thirdparty/multierr"
 )
 
-// Open listeners for each network the swarm should listen on
-func (s *Swarm) listen(addrs []ma.Multiaddr) error {
-
+// Open listeners and reuse-dialers for the given addresses
+func (s *Swarm) setupAddresses(addrs []ma.Multiaddr) error {
 	for _, addr := range addrs {
 		if !addrutil.AddrUsable(addr, true) {
 			return fmt.Errorf("cannot use addr: %s", addr)
 		}
 	}
 
-	retErr := multierr.New()
+	dialer := manet.Dialer{Dialer: net.Dialer{Timeout: DialTimeout}}
+	for _, a := range addrs {
+		switch {
```

Comment:
I thought that the abstraction was that the callers of the swarm didnt need to care about what the addresses were. My bad, i'll move it elsewhere.